lib/puppet/type/cisco_interface.rb

```diff
@@ -698,6 +702,206 @@ def is_to_s(value)
     end
   end # property stp_vlan_port_priority
 
+  ###########################
+  # private vlan attributes #
+  ###########################
+
+  newproperty(:switchport_mode_private_vlan_host) do
+    desc 'Switchport private host mode of the interface.'
+
+    newvalues(
+      :host,
+      :promiscuous,
+      :disabled)
+  end # property switchport_mode_private_vlan_host
+
+  newproperty(:switchport_mode_private_vlan_host_association, array_matching: :all) do
+    format = '["primary_vlan", "secondary_vlan"]'
+    desc 'An array of [primary_vlan, secondary_vlan] pair. '\
+         "Valid values match format #{format} with primary_vlan "\
+         'and secondary_vlan as integer string.'
+    match_error = "must be of format #{format} with primary_vlan and "\
+                  'secondary_vlan as integer string.'
+
+    validate do |value|
```

Comment:
General notes about the validate methods for arrays.
- Users should be able to specify both integers(fixnums) or strings that represent integers.
- You also need to handle `'default'`.

I would suggest the following:
```ruby
validate do |vlans|
  vlans.split.each do |vlan|
    fail "Vlan value '#{vlan}' #{match_error}" unless /^\d+/.match)vlan.to_s ||
      value == 'default' || value = :default
  end
end